src/test/java/org/terasology/moduletestingenvironment/fixtures/ItemDropSpyTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.moduletestingenvironment.fixtures;
+
+import com.google.common.collect.Sets;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.inventory.events.DropItemEvent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.*;
+
+public class ItemDropSpyTest extends ModuleTestingEnvironment {
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Override
+  public Set<String> getDependencies() {
+    return Sets.newHashSet("engine", "ModuleTestingEnvironment");
+  }
+
+  @Test
+  public void cumulativeDropsTest() {
+    final List<EntityRef> referenceDrops = new ArrayList<>();
+    try (ItemDropSpy spy = new ItemDropSpy(getHostContext())) {
+      List<EntityRef> drops = spy.getDrops();
+      assertTrue(drops.isEmpty());
+      for (int i = 0; i < 5; i++) {
+        referenceDrops.add(dropItem());
+        assertEquals(i + 1, drops.size());
+        assertEquals(referenceDrops.get(i), drops.get(i));
+      }
+    }
+  }
+
+  @Test
+  public void properClosureTest() {
+    List<EntityRef> drops;
+    try (ItemDropSpy spy = new ItemDropSpy(getHostContext())) {
+      drops = spy.getDrops();
+    }
+    dropItem();
+    assertTrue(drops.isEmpty());
+  }
+
+  @Test
+  public void staticUsageTest() {
+    /* We need to wrap this forward declaration in an atom so it can be final and the lambda will
```

Comment:
I think this could just be a regular // comment - our suggested max line length is quite a bit longer than this one so multi-line shouldn't be needed :-)

Reply:
It looks like official max line length is 175, which I find a little absurd.  I've been trying to keep it under 100, since my window width is only about 102 when I have any of the left-side tool windows open.  (I typically run two side-by-side windows.)

At any rate, the change to indents put this one at 126, but I reworded it so it's only 112.  I probably should have done that to begin with; I agree it was a little ugly.

Note that I've been wrapping the JavaDocs to 100, but my narrow windows aside, I think there's a solid case for wrapping English more aggressively than code.

Reply:
Yeah the line length is never going to be perfect for every case, and at this point I've forgotten how we got so high :D Good solution to just tighten up the comment a bit!